haystack/components/preprocessors/utils.py

```diff
@@ -0,0 +1,228 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import re
+from pathlib import Path
+from typing import Any, Dict, List, Literal, Tuple
+
+from haystack import logging
+from haystack.lazy_imports import LazyImport
+
+with LazyImport("Run 'pip install nltk'") as nltk_imports:
+    import nltk
+
+nltk_imports.check()
+
+logger = logging.getLogger(__name__)
+
+Language = Literal[
+    "ru", "sl", "es", "sv", "tr", "cs", "da", "nl", "en", "et", "fi", "fr", "de", "el", "it", "no", "pl", "pt", "ml"
+]
+ISO639_TO_NLTK = {
+    "ru": "russian",
+    "sl": "slovene",
+    "es": "spanish",
+    "sv": "swedish",
+    "tr": "turkish",
+    "cs": "czech",
+    "da": "danish",
+    "nl": "dutch",
+    "en": "english",
+    "et": "estonian",
+    "fi": "finnish",
+    "fr": "french",
+    "de": "german",
+    "el": "greek",
+    "it": "italian",
+    "no": "norwegian",
+    "pl": "polish",
+    "pt": "portuguese",
+    "ml": "malayalam",
+}
+
+
+class CustomPunktLanguageVars(nltk.tokenize.punkt.PunktLanguageVars):
+    # The following adjustment of PunktSentenceTokenizer is inspired by:
+    # https://stackoverflow.com/questions/33139531/preserve-empty-lines-with-nltks-punkt-tokenizer
+    # It is needed for preserving whitespace while splitting text into sentences.
+    _period_context_fmt = r"""
+        %(SentEndChars)s             # a potential sentence ending
+        \s*                          # match potential whitespace [ \t\n\x0B\f\r]
+        (?=(?P<after_tok>
+            %(NonWord)s              # either other punctuation
+            |
+            (?P<next_tok>\S+)        # or some other token - original version: \s+(?P<next_tok>\S+)
+        ))"""
+
+    def period_context_re(self) -> re.Pattern:
+        """
+        Compiles and returns a regular expression to find contexts including possible sentence boundaries.
+
+        :returns: A compiled regular expression pattern.
+        """
+        try:
+            return self._re_period_context  # type: ignore
+        except:  # noqa: E722
+            self._re_period_context = re.compile(
+                self._period_context_fmt
+                % {
+                    "NonWord": self._re_non_word_chars,
+                    # SentEndChars might be followed by closing brackets, so we match them here.
+                    "SentEndChars": self._re_sent_end_chars + r"[\)\]}]*",
+                },
+                re.UNICODE | re.VERBOSE,
+            )
+            return self._re_period_context
+
+
+def load_sentence_tokenizer(
+    language: Language, keep_white_spaces: bool = False
+) -> nltk.tokenize.punkt.PunktSentenceTokenizer:
+    """
+    Utility function to load the nltk sentence tokenizer.
+
+    :param language: The language for the tokenizer.
+    :param keep_white_spaces: If True, the tokenizer will keep white spaces between sentences.
+    :returns: nltk sentence tokenizer.
+    """
+    try:
```

Comment:
What's the right thing to use here? See the issue described in https://github.com/deepset-ai/haystack/issues/8238 (and the corrective action recommended) 
In 1.26.x branch we load these nltk thingies a bit differently, see these [LOCs](https://github.com/deepset-ai/haystack/blob/v1.26.x/haystack/nodes/preprocessor/preprocessor.py#L120)

Reply:
Ok keeping it as is